lib-extra/src/main/java/com/diffplug/spotless/extra/config/EclipseConfiguration.java

```diff
@@ -0,0 +1,257 @@
+		for (String allowedVersion : supportedEclipseVersions) {
+			this.supportedEclipseVersions.add(new SemanticVersion(allowedVersion));
+		}
+		version = this.supportedEclipseVersions.last(); //Use latest version per default
+		defaultCoordinatesUrl = getDefaultCoordinatesUrl(eclipseConfigContext, version);
+		this.jarProvisioner = jarProvisioner;
+
+		coordinateAdaptations = new MavenCoordinates(); //Use default coordinates configured for version
+		settingsFiles = new ArrayList<File>(); //Use default preferences
+		state = null; //Create state only on demand too speed up initialization
+	}
+
+	/** Set Eclipse version */
+	public void setVersion(String versionOrUrl) {
+		try {
+			// For development purpose and patch delivery it is allowed
+			// to specify directly the URL of a M2 coordinates file.
+			defaultCoordinatesUrl = new URL(versionOrUrl);
+			// The default version is not changed.
+		} catch (MalformedURLException ignored) {
+			SemanticVersion newVersion = new SemanticVersion(versionOrUrl);
+			if (supportedEclipseVersions.contains(newVersion)) {
+				defaultCoordinatesUrl = getDefaultCoordinatesUrl(eclipseConfigContext, newVersion);
+				version = newVersion;
+			} else {
+				throw new UserArgumentException(versionOrUrl,
+						String.format("Version is not part of the supported versions '%s'.",
+								supportedEclipseVersions.stream().map(v -> v.toString()).collect(Collectors.joining(", "))));
+			}
+		}
+		invalidateState();
+	}
+
+	private static URL getDefaultCoordinatesUrl(URL eclipseConfigContext, SemanticVersion version) {
+		try {
+			Path filePath = Paths.get(eclipseConfigContext.getPath(), String.format("v%s.txt", version.toString()));
+			return new URL(eclipseConfigContext.getProtocol(), eclipseConfigContext.getHost(), filePath.toString());
+		} catch (MalformedURLException e) {
+			/*
+			 *  This exception must be prevented by the strict syntax checking of
+			 *  SemanticVersion and unit-tests of all allowed versions.
+			 */
+			throw new RuntimeException(e);
+		}
+	}
+
+	/**
+	 * Modify default Maven dependencies configured for the formatter.
+	 * The default dependencies are located in {@link #ECLIPSE_FORMATTER_RESOURCES}.
+	 */
+	public void setDepenencies(String... coordinates) {
+		coordinateAdaptations = new MavenCoordinates();
+		coordinateAdaptations.update(coordinates);
+		invalidateState();
+	}
+
+	/** Set settings files containing Eclipse preferences */
+	public void setPreferences(Iterable<File> settingsFiles) {
+		this.settingsFiles = settingsFiles;
+		invalidateState();
+	}
+
+	/** The current state becomes invalid in case a configuration item has successfully changed */
+	private void invalidateState() {
+		state = null;
+	}
+
+	/** Creates the state of the configuration */
+	public EclipseConfiguration.State get() {
+		if (null == state) {
+			state = new State(
+					jarProvisioner,
+					createMavenCoordinates(),
+					version,
+					settingsFiles);
+		}
+		return state;
```

Comment:
Don't know who and how often `get()` is called. Since this code is not `gradle` specific, I wanted to be on the save side. I do access here files from the resources, so the procedure costs some time, though not very much. Anyhow, I have no strong feelings about this. Let me know you final decision.

Reply:
> There are 2 hard problems in computer science: cache invalidation, naming things, and off-by-1 errors.

:-P  I think the safe thing is to avoid caching until it's absolutely necessary.